json/src/main/java/org/jdbi/v3/json/JsonMapper.java

```diff
@@ -26,6 +26,19 @@
  * jdbi3-jackson2 and jdbi3-gson2 are readily available for this.
  */
 public interface JsonMapper {
-    String toJson(Type type, Object value, ConfigRegistry config);
-    Object fromJson(Type type, String json, ConfigRegistry config);
+    @Deprecated // forRemoval=true
+    default String toJson(Type type, Object value, ConfigRegistry config) {
+        return forType(type, config).toJson(value, config);
+    }
+    @Deprecated // forRemoval=true
+    default Object fromJson(Type type, String json, ConfigRegistry config) {
+        return forType(type, config).fromJson(json, config);
+    }
+
+    TypedMapper forType(Type type, ConfigRegistry config);
+
+    interface TypedMapper {
```

Comment:
JsonTypedMapper ? "TypedMapper" is too generic for my taste. 

This is basically a 

```java
public interface SerDe<T> {
    String serialize(T value, ConfigRegistry config);
    T deserialize(String value, ConfigRegistry config);
}
```

This may be useful in other contexts as well.

Reply:
Sure, it is really a nested type, so the name is `JsonMapper.TypedMapper` which already feels a bit redundant and `JsonMapper.JsonTypedMapper` feels even more redundant. But I do not care too much.

I am happy for the interface to be re-used but I do not see another place for it to be used right now.